sql-plugin/src/main/scala/com/nvidia/spark/rapids/spill/SpillFramework.scala

```diff
@@ -177,14 +178,44 @@ trait SpillableHandle extends StoreHandle {
    */
   def spill(): Long
 
+  private val spilling = new AtomicBoolean(false)
+
+  /**
+   * Method used to atomically check and set the spilling state, so that anyone who wants to
+   * actually perform a spill can ensure they are the only one spilling, without having to block
+   * on the actual spill operation (IO). Only someone who has set spilling to true to perform their
+   * spill may set it back to false when they are done. (Visible for tests)
+   *
+   * This is a separate check from spillable, which actually checks the state of the buffer handle
+   *
+   * @param s whether the caller is trying to spill or not (ie finished)
+   * @return whether the caller is allowed to spill (or true if s is false)
+   */
+  def setSpilling(s: Boolean): Boolean = {
```

Comment:
nit: Could we split this up into two separate APIs?  There is never a case where we programatically pass in `s` and we use `s` only to change from one implementation to another.  I think it would make the code cleaner to read also.